evaluation/evaluator.go

```diff
@@ -400,67 +402,75 @@ func (e Evaluator) getVariationForTheFlag(flag *rest.FeatureConfig, target *Targ
 }
 
 // BoolVariation returns boolean evaluation for target
-func (e Evaluator) BoolVariation(identifier string, target *Target, defaultValue bool) bool {
+func (e Evaluator) BoolVariation(identifier string, target *Target, defaultValue bool) (bool, error) {
```

Comment:
@davejohnston same as my previous PR for the public client XVariation methods. 

We now return the error from the evaluator to the client, so we can bubble it back up to the user. 

What we did in the previous PR was just if the client wasn't initialized. Now this will cover the scenario when the evaluator has to return an error